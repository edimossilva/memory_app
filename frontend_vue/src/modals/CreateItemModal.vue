<template>
  <modal name="create-item-modal" class="create-item-modal-js">
    <label> Key: </label>
    <input
      v-model="item.key"
      type="text"
      class="create-item-modal--input-key-js"
    />
    <label> Value: </label>
    <input
      v-model="item.value"
      type="text"
      class="create-item-modal--input-value-js"
    />
    <btn
      label="create"
      :onClick="onCreateButtonClick"
      class="create-item-modal--button-create-js"
    ></btn>
  </modal>
</template>

<script>
import Btn from "@/components/Btn.vue";
import { mapActions } from "vuex";

export default {
  name: "CreateItemModal",
  components: { Btn },
  data() {
    return {
      item: { key: "", value: "" },
    };
  },
  methods: {
    ...mapActions(["addItem"]),
    onCreateButtonClick() {
      this.addItem(this.item);
      this.$modal.hide("create-item-modal");
      this.item = {};
    },
  },
};
</script>
