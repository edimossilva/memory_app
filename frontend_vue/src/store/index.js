import Vue from "vue";
import Vuex from "vuex";

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    items: [],
  },
  mutations: {
    addItem(state, item) {
      state.items.push(item);
    },
    removeItem(state, item) {
      const index = state.items.findIndex((i) => i.key === item.key);
      state.items.splice(index, 1);
    },
    setItems(state, items) {
      state.items = items;
    },
    // mutate(state, payload) {
    //   state[payload.property] = payload.value;
    // },
  },
  actions: {
    addItem({ commit }, item) {
      commit("addItem", { ...item });
    },
    removeItem({ commit }, item) {
      commit("removeItem", item);
    },
    setItems({ commit }, items) {
      commit("setItems", items);
    },
  },
  modules: {},
});
