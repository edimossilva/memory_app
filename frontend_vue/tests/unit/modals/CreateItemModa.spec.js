import { shallowMount } from "@vue/test-utils";
import CreateItemModal from "@/modals/CreateItemModal.vue";
import Vue from "vue";
import VueRouter from "vue-router";
import store from "../../../src/store/index";

Vue.use(VueRouter);
describe("CreateItemModal.vue", () => {
  const spyUpdate = jest.spyOn(CreateItemModal.methods, "addItem");

  it("Matchs Snapshot", () => {
    const wrapper = shallowMount(CreateItemModal, {
      stubs: {
        modal: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  describe("methods", () => {
    describe("onCreateButtonClick()", () => {
      const $modal = { hide: jest.fn() };
      const wrapper = shallowMount(CreateItemModal, {
        store,
        stubs: {
          modal: true,
        },
        mocks: {
          $modal,
        },
      });

      beforeEach(() => {
        wrapper.vm.onCreateButtonClick();
      });

      it("should call push to dashboard", () => {
        expect(spyUpdate).toHaveBeenCalled();
      });
    });
  });
});
