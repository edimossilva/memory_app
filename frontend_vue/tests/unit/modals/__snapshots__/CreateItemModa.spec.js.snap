// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateItemModal.vue Matchs Snapshot 1`] = `
<modal-stub name="create-item-modal" class="create-item-modal-js"><label> Key: </label> <input type="text" class="create-item-modal--input-key-js"> <label> Value: </label> <input type="text" class="create-item-modal--input-value-js">
  <btn-stub label="create" onclick="function () { [native code] }" class="create-item-modal--button-create-js"></btn-stub>
</modal-stub>
`;
